src/Security/Security.php

```diff
@@ -206,15 +213,120 @@ class Security extends Controller implements TemplateGlobalProvider
      * @var boolean If set to TRUE or FALSE, {@link database_is_ready()}
      * will always return FALSE. Used for unit testing.
      */
-    static $force_database_is_ready = null;
+    protected static $force_database_is_ready = null;
 
     /**
      * When the database has once been verified as ready, it will not do the
      * checks again.
      *
      * @var bool
      */
-    static $database_is_ready = false;
+    protected static $database_is_ready = false;
+
+    /**
+     * @var array available authenticators
+     */
+    protected static $authenticators = [];
```

Comment:
Config must be private static not protected static.
However, I've suggested this should be changed from a config var to an injected dependency, so shall we type this as @var Authenticator[] and inject these instead?